EpikCord/__init__.py

```diff
@@ -1891,7 +1894,93 @@ def unload_section(self, section: Union[CommandsSection, EventsSection]):
 # class ClientGuildMember(Member):
 #     def __init__(self, client: Client,data: dict):
 #         super().__init__(data)
+class VoiceWebsocketClient:
+    def __init__(self, client:Client):
+        self.client = client
+        self.HELLO = 8
+        self.IDENTIFY = 0
+        self.HEARTBEAT = 3
+        self.sequence = None
+
+    async def connect(self, guild_id:str, channel_id:str, **kwargs):
+        self.guild_id = guild_id
+        await self.client.send_json({
+            "op": 4,
+            "d" : {
+                "guild_id": guild_id,
+                "channel_id": channel_id,
+                "self_mute": kwargs.get("self_mute", False),
+                "self_deaf" : kwargs.get("self_deaf", False)
+            }
+        })
+        def figure_out_receive(data):
+            if data["t"] == "VOICE_SERVER_UPDATE":
+                return "VOICE_SERVER_UPDATE"
+            elif data["deaf"]:
+                return "VOICE_STATE_UPDATE"
+            else:
+                return ""
+        first_receive, second_receive = await asyncio.gather(await self.client.ws.receive(),await self.client.ws.receive())
+        receive1 = first_receive.json()
+        receive2 = second_receive.json()
+        if figure_out_receive(receive1) == "VOICE_SERVER_UPDATE":
+            voice_server_update=receive1
+        elif figure_out_receive(receive2) == "VOICE_SERVER_UPDATE":
+            voice_server_update=receive2
+        
+        if figure_out_receive(receive1) == "VOICE_STATE_UPDATE":
+            self.session_id=receive1["session_id"]
+        elif figure_out_receive(receive2) == "VOICE_STATE_UPDATE":
+            self.session_id=receive2["session_id"]
+
+        print(self.session_id)
+        print(voice_server_update)
+        self.token = voice_server_update["token"]
+        self.endpoint = voice_server_update["endpoint"]
+        
+        self.ws = await self.client.http.ws_connect("wss://"+ self.endpoint + "/?v=4")
+        await self.identify()
+        ready_resp = await self.ws.receive()
+        ready_parsed_data = await ready_resp.json()["d"]
+        self.ssrc:int = ready_parsed_data["ssrc"]
+        self.ip:str = ready_parsed_data["ip"]
+        self.port:int = ready_parsed_data["port"]
+        self.encryption_modes:list = ready_parsed_data["modes"]
+        self.heartbeat_interval:int = (await self.ws.receive()).json()["d"]["heartbeat_interval"] #hello my friend, how ya doing. Gimme the delay after which I tell i am still alive?
+        async def heartbeat():
+            while True:
+                await self.heartbeat(False)
+        asyncio.create_task(heartbeat())
+
+    async def send_json(self, json):
+        '''Sends JSON data to the ``voice`` websocket connection
+        Note: Fails when used before ``connect()``'''
```

Comment:
If you're going to tell us when it fails, add in if it's disconnected.

Reply:
OK :)